### PR TITLE
signal: Backtrace on SIGUSR1

### LINKED ISSUE #66: Backtrace on SIGUSR1
Implement same logic as on https://github.com/kata-containers/runtime/issues/241.

### PR DESCRIPTION
**Pull request by jodh-intel**
Rework the signal handling code so that if debug is enabled and a
`SIGUSR1` signal is received, backtrace to the system log but continue
to run.

Added some basic tests for the signal handling code.

Fixes #66.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: signals.go [hunk 1]
```diff
@@ -55,15 +58,28 @@ func backtrace() {
...
+	return handledSignalsMap[sig]
+}
+
+func nonFatalSignal(sig syscall.Signal) bool {
+	_, exists := handledSignalsMap[sig]
+	if !exists {
+		return false
+	}
+
+	return !handledSignalsMap[sig]
```
Comment (devimc) [inline signals.go]: I think you can improve the performance just a little , see https://github.com/kata-containers/runtime/pull/244  :smile:

File: proxy.go [hunk 2]
```diff
@@ -196,31 +198,54 @@ func printAgentLogs(sock string) error {
...
 	}
 
-	return sigCh
+	go func() {
+		for {
+			// Block waiting for a signal
+			sig := <-sigCh
+
+			nativeSignal, ok := sig.(syscall.Signal)
+			if ok {
```
[the remark this answers is not in the document]
Comment (jodh-intel) [reply proxy.go]: Also fixed on https://github.com/kata-containers/runtime/pull/244.

**Comment by sboeuf** [comment]
@bergwolf PTAL, this is ready to be merged.

**Comment by jodh-intel** [comment]
Hi @bergwolf - ptal. I'd really like the `SIGUSR1` handling landed for all components ;)

**Comment by jodh-intel** [comment]
Green button... must push it!

**Pull request merged**